mm2src/mm2_main/src/rpc.rs

```diff
@@ -175,12 +175,21 @@ fn response_from_dispatcher_error(
     response.serialize_http_response()
 }
 
-async fn process_single_request(ctx: MmArc, req: Json, client: SocketAddr) -> Result<Response<Vec<u8>>, String> {
+async fn process_single_request(ctx: MmArc, mut req: Json, client: SocketAddr) -> Result<Response<Vec<u8>>, String> {
     let local_only = ctx.conf["rpc_local_only"].as_bool().unwrap_or(true);
     if req["mmrpc"].is_null() {
-        return dispatcher_legacy::process_single_request(ctx, req, client, local_only)
-            .await
-            .map_err(|e| ERRL!("{}", e));
+        match dispatcher_legacy::process_single_request(ctx.clone(), req.clone(), client, local_only).await {
+            Ok(t) => return Ok(t),
+
+            Err(dispatcher_legacy::LegacyRequestProcessError::NoMatch) => {
+                // Try the v2 implementation
+                req["mmrpc"] = json!("2.0");
+            },
+
+            Err(e) => {
+                return ERR!("{}", e);
+            },
+        };
```

Comment:
you can create a const for `mmrpc` inside this fn and reuse ...
```rust
const MM_RPC: &str = "mmrpc";
```

Reply:
I stand to be corrected, with the code above, won't `mmrpc` be generated at the different locations it's called(I mean the compiler can optimized this away but not everytime)? unlike if we created a single const then the compiler with just substitute

Reply:
"mmrpc" here is static string, so it's not runtime generated stuff.